packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -604,4 +617,55 @@ contract PluginSetupProcessor is DaoAuthorizable {
             });
         }
     }
+
+    /// @notice Checks the update parameters
+    /// @param _updateParams The parameters of the update.
+    /// @return oldVersion The old semantic version number being needed in the context for further usage.
+    function _checkUpdateValidity(PluginUpdateParams calldata _updateParams)
+        internal
+        view
+        returns (uint16[3] memory oldVersion)
+    {
+        // These revert if the current or new plugin setup are not part of the plugin setup repo
+        (oldVersion, , ) = _updateParams.pluginSetupRepo.getVersionByPluginSetup(
+            _updateParams.currentPluginSetup
+        );
+        (uint16[3] memory newVersion, , ) = _updateParams.pluginSetupRepo.getVersionByPluginSetup(
+            _updateParams.newPluginSetup
+        );
+
+        // Assert that the version bump valid
+        if (!isValidUpdate(oldVersion, newVersion)) {
+            revert UpdateInvalid({currentVersion: oldVersion, invalidNextVersion: newVersion});
+        }
+    }
+
+    /// @notice Checks if an update is valid by the version numbers.
+    /// @param _oldVersion The old semantic version number.
+    /// @param _newVersion The new semantic version number.
+    /// @return bool Returns true if the bump is valid.
+    function isValidUpdate(uint16[3] memory _oldVersion, uint16[3] memory _newVersion)
+        public
+        pure
+        returns (bool)
+    {
+        bool hasIncreased;
+        uint256 i = 0;
+        while (i < 3) {
+            if (hasIncreased) {
+                if (_newVersion[i] != 0) {
+                    return false;
+                }
+            } else if (_newVersion[i] != _oldVersion[i]) {
+                if (_oldVersion[i] > _newVersion[i]) {
```

Comment:
This function is a copy/paste of `isValidBump()` of PluginRepo.sol except for one of the conditions.

Perhaps we can find a way to share the function, for example:

`function isValidBump(
        uint16[3] memory _oldVersion,
        uint16[3] memory _newVersion,
        bool strict
    ) public pure returns (bool) {
        bool hasBumped;
        uint256 i = 0;
        while (i < 3) {
            if (hasBumped) {
                if (_newVersion[i] != 0) {
                    return false;
                }
            } else if (_newVersion[i] != _oldVersion[i]) {
                if (
                    _oldVersion[i] > _newVersion[i] ||
                    (strict ? _newVersion[i] - _oldVersion[i] != 1 : true)
                ) {
                    return false;
                }
                hasBumped = true;
            }
            i = _uncheckedIncrement(i);
        }
        return hasBumped;
    }`
    
    so we can use strict check on PluginRepo, and non-strict here for example
    it is just a suggestion

Reply:
See https://github.com/aragon/core/pull/130#discussion_r1014013408